### PR TITLE
[Merged by Bors] - Fix axis settings constructor

### PR DESCRIPTION
**Pull request by nicopap**
# Objective

Currently, the `AxisSettings::new` function is unusable due to
an implementation quirk. It only allows `AxisSettings` where
the bounds that are supposed to be positive are negative!

## Solution

- We fix the bound check
- We add a test to make sure the method is usable

Seems like the error slipped through because of the relatively
verbose code style. With all those `if/else`, very long names,
range syntax, the bound check is actually hard to spot. I first
refactored a lot of code, but I left out the refactor because the
fix should be integrated independently.

---

## Changelog

- Fix `AxisSettings::new` only accepting invalid bounds

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]
I'd love to see some simple tests here, but I won't block on it.

Refactoring would also be great, but yes, a seperate PR is better.

File: crates/bevy_input/src/gamepad.rs [hunk 1]
```diff
@@ -638,19 +659,21 @@ impl AxisSettings {
         livezone_upperbound: f32,
         threshold: f32,
     ) -> Result<AxisSettings, AxisSettingsError> {
-        if !(-1.0..=0.0).contains(&livezone_lowerbound) {
+        let within = |value, lower: f32, upper: f32| (lower..=upper).contains(&value);
```
Comment (rparrett) [inline crates/bevy_input/src/gamepad.rs:662]: I think this should also be moved into a separate refactor PR if that is coming. I personally feel like this is not making the code more readable, but that would be a better place to discuss.

**Comment by nicopap** [comment]
I also removed `new_symmetric` from the PR, it didn't belong.

**Review by james7132 — APPROVED** [review]
bors r+

**Pull request closed without merging**